pkg/shifter/shifter.go

```diff
@@ -0,0 +1,255 @@
+package shifter
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/roppenlabs/silent-assassin/pkg/config"
+	"github.com/roppenlabs/silent-assassin/pkg/gcloud"
+	"github.com/roppenlabs/silent-assassin/pkg/k8s"
+	"github.com/roppenlabs/silent-assassin/pkg/killer"
+	"github.com/roppenlabs/silent-assassin/pkg/logger"
+	"github.com/roppenlabs/silent-assassin/pkg/notifier"
+	container "google.golang.org/api/container/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+const timeLayout string = "15:04"
+
+//Suggest a better name
+type npShiftConf struct {
+	destNodePool       string
+	sourceMinNodeCount int64
+}
+
+type ShifterService struct {
+	cp                 config.IProvider
+	logger             logger.IZapLogger
+	kubeClient         k8s.IKubernetesClient
+	gcloudClient       gcloud.IGCloudClient
+	killer             killer.KillerService
+	notifier           notifier.INotifierClient
+	nodePoolMap        map[string]npShiftConf
```

Comment:
This can come as a result of the getNodePoolMap function. We don't need it to initialise the shifter service.